pkg/controllers/state/suite_test.go

```diff
@@ -523,6 +523,38 @@ var _ = Describe("Inflight Nodes", func() {
 		// Still expect to succeed w/o instance type since inflight details are set once
 		ExpectReconcileSucceeded(ctx, nodeController, client.ObjectKeyFromObject(node))
 	})
+	It("should populate node from the cluster if the node initialized", func() {
+		instanceType := cloudProvider.InstanceTypes[0]
```

Comment:
I think we want to enforce the scenario we know that can happen here, which is that the GetInstanceTypes() call returns no valid instance types, but we still want to succeed to propagate resources. There's an `InstanceTypes` part of the fake cloudprovider that you could use to assign no instance types to the `GetInstanceTypes()` call so that you ensure that this is properly tested.